gorm/filtering_test.go

```diff
@@ -216,6 +216,41 @@ func TestGormFiltering(t *testing.T) {
 			nil,
 			nil,
 		},
+		{
+			"id := 'ID'",
+			"(lower(entities.id) = lower(?))",
+			[]interface{}{"convertedid"},
+			nil,
+			nil,
+		},
+		{
+			"(id := 'sOmeId') and (ref := 'sOmeRef')",
+			"((lower(entities.id) = lower(?)) AND (lower(entities.ref) = lower(?)))",
+			[]interface{}{"convertedid", "convertedref"},
+			nil,
+			nil,
+		},
+		{
+			"not(id := 'sOmeId') and not(ref := 'sOmeRef')",
+			"(NOT(lower(entities.id) = lower(?)) AND NOT(lower(entities.ref) = lower(?)))",
+			[]interface{}{"convertedid", "convertedref"},
+			nil,
+			nil,
+		},
+		{
+			"(not(id in ['sOmeId', 'egegeg']) and (not(field := 'Hello')))",
+			"((entities.id NOT IN (?, ?)) AND NOT(lower(field) = lower(?)))",
+			[]interface{}{"convertedid", "convertedid", "Hello"},
+			nil,
+			nil,
+		},
+		{
+			"(not(field2 in ['sOmeId', 'egegeg']) and (not(field := 'Hello')))",
+			"((entities.field2 NOT IN (?, ?)) AND NOT(lower(field) = lower(?)))",
+			[]interface{}{"sOmeId", "egegeg", "Hello"},
+			nil,
+			nil,
+		},
```

Comment:
Need a test case for `ArrayNumberCondition` as well - not a complex expression, just a simple one.